tests/functional/conftest.py

```diff
@@ -1,22 +1,48 @@
 import logging
+import pathlib
 import tempfile
 import time
 import uuid
-from pathlib import Path
 from subprocess import check_output
 
 import pytest
+import requests
 
 import gitlab
 import gitlab.base
 from tests.functional import helpers
 
+SLEEP_TIME = 20
```

Comment:
Could we try with 10 sec initially? I've been using this simple check myself for another project and it seems to be ok:

```python
@pytest.fixture(scope="session")
def service_ready(gitlab_url: str) -> None:
    for _ in GITLAB_MAX_ATTEMPTS:
        try:
            response = requests.get(f"{gitlab_url}/-/health")
            assert "GitLab OK" in getattr(response, "text", "")
        except (requests.ConnectionError, AssertionError):
            sleep(10)
            continue

        return

    raise TimeoutError("GitLab health checks timed out")
```

Reply:
Okay. I guess you like to live on the edge of danger! I prefer a larger margin of error...

Reply:
I'm probably biased as I spent too much time optimizing this from a hardcoded 300 seconds wait a while back :sweat_smile: